external-crates/move/crates/move-compiler/src/sui_mode/linters/freezing_capability.rs

```diff
@@ -90,27 +90,26 @@ impl<'a> TypingVisitorContext for Context<'a> {
     }
 }
 
-impl<'a> Context<'a> {
-    fn is_freeze_function(&self, fun: &T::ModuleCall) -> bool {
-        FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
-            fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
-        })
-    }
-
-    fn check_type_arguments(&mut self, fun: &T::ModuleCall, loc: Loc) {
-        for sp!(_, type_arg) in &fun.type_arguments {
-            if let Some(sp!(_, TypeName_::ModuleType(_, struct_name))) = type_arg.type_name() {
-                if self.capability_regex.is_match(struct_name.value().as_str()) {
-                    self.report_freeze_capability(loc);
-                    break;
-                }
-            }
-        }
-    }
+fn is_freeze_function(fun: &T::ModuleCall) -> bool {
+    FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
+        fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
```

Comment:
Would it be cheaper to convert the names into symbols to avoid the `as_str()` check?

Reply:
Maybe, but a lot of these aren't static symbols already, so not sure it would be worth it 🤷